helix-term/src/commands/typed.rs

```diff
@@ -1388,9 +1413,11 @@ fn sort_impl(
         .map(|fragment| fragment.chunks().collect())
         .collect();
 
-    fragments.sort_by(match reverse {
-        true => |a: &Tendril, b: &Tendril| b.cmp(a),
-        false => |a: &Tendril, b: &Tendril| a.cmp(b),
+    fragments.sort_by(match (reverse, alphanumeric) {
+        (true, false) => |a: &Tendril, b: &Tendril| b.cmp(a),
+        (false, false) => |a: &Tendril, b: &Tendril| a.cmp(b),
+        (true, true) => |a: &Tendril, b: &Tendril| numeric_cmp(b, a),
+        (false, true) => |a: &Tendril, b: &Tendril| numeric_cmp(a, b),
     });
```

Comment:
I agree with @leandrobbraga - this block is a little dense and we can make the call sites for `sort_impl` a little cleaner by moving these closures to parameters of `sort_impl`.

So this function would replace the `reverse` parameter with a `compare_fn: impl FnMut(&Tendril, &Tendril) -> std::cmp::Ordering,` param and then this block can be just `fragments.sort_by(compare_fn);`.

We could also use this opportunity to drop the `args` parameter which is currently unused